pkg/eval/testutils.go

```diff
@@ -120,6 +120,28 @@ func (e errWithMessage) matchError(e2 error) bool {
 	return e2 != nil && e.msg == e2.Error()
 }
 
+// An errorMatcher for an ExternalCmdExit error that ignores the `Pid` member.
+// We only match the command name and exit status because at run time we
+// cannot know the correct value for `Pid`.
+type errCmdExit struct{ v ExternalCmdExit }
+
+func (e errCmdExit) Error() string {
+	return e.v.Error()
+}
+
+func (e errCmdExit) matchError(gotErr error) bool {
+	if gotErr == nil {
+		return false
+	}
+
+	ge, ok := gotErr.(*Exception).Reason.(ExternalCmdExit)
+	if !ok {
+		panic(fmt.Sprintf("errCmdExit.matchError expected eval.ExternalCmdExit, got %T", gotErr))
+	}
```

Comment:
Since this error is handled by a panic anyway, use the single-lhs form of type assertion, which panics when the type doesn't match.

Reply:
Okay. I did this in the hope the more explicit panic message might help but since it should never be seen I can see your point that there is little benefit to those lines.